src/components/pages/Home/Onboarding/Stepper.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { ReactElement } from 'react'
+import styles from './Stepper.module.css'
+import classNames from 'classnames/bind'
+import Button from '../../../atoms/Button'
+import { NavigationDirections } from '.'
+import { useWeb3 } from '../../../../providers/Web3'
+import { toast } from 'react-toastify'
+import { getErrorMessage } from '../../../../utils/onboarding'
+
+const cx = classNames.bind(styles)
+
+export default function Stepper({
+  stepLabels,
+  currentStep,
+  onboardingCompleted,
+  setCurrentStep,
+  setNavigationDirection
+}: {
+  stepLabels: string[]
+  currentStep: number
+  onboardingCompleted: boolean
+  setCurrentStep: (step: number) => void
+  setNavigationDirection: (direction: NavigationDirections) => void
+}): ReactElement {
+  const { accountId, balance, networkId, web3Provider } = useWeb3()
+
+  const handleClick = (newStep: number) => {
+    if (newStep === stepLabels.length - 1 && !onboardingCompleted) {
+      toast.error(
+        getErrorMessage({
+          accountId,
+          web3Provider: !!web3Provider,
+          networkId,
+          balance
+        })
+      )
+      return
+    }
+
+    currentStep > newStep
+      ? setNavigationDirection(NavigationDirections.PREV)
+      : setNavigationDirection(NavigationDirections.NEXT)
+
+    setCurrentStep(newStep)
+  }
+  return (
+    <div>
+      <ol className={styles.stepper}>
+        {stepLabels.map((label, i) => (
+          <li
+            key={i}
+            className={cx({
+              step: true,
+              active: i <= currentStep
+            })}
+          >
+            <Button
+              style="text"
+              className={styles.title}
+              onClick={() => handleClick(i)}
```

Comment:
thought/nitpick(non-blocking): usability / accessibility of this component would be imrpoved if the "step circles" could also be clicked imo

Reply:
We thought about it but the connecting lines between the circles are part of the same component and they would be clickable as well, not sure I have a solution atm to make only the circles clickable

Reply:
Can be an improvement in a later version if needed +1